File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/login/MemoryRepository.java
package com.charlye934.loginddagger.login;

public class MemoryRepository implements LoginRepository {

    private User user;

    @Override
    public void saveUser(User user) {
        if(user == null){
            user = getUser();
        }

        this.user = user;
    }

    @Override
    public User getUser() {
        if(user == null){
            user = new User("Carlos","Arteaga");
            return user;
        }else {
            return user;
        }
    }
}

File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/login/MemoryRepository.kt
package com.charlye934.loginddagger.login

class MemoryRepository : LoginRepository{

    private var user: User? = null

    override fun saveUser(user: User) {
        if(this.user == null){
            this.user = getUser()
        }

        this.user = user
    }

    override fun getUser(): User {
        return if(user == null){
            user = User("Carlos", "Arteaga")
            user as User
        }else{
            user as User
        }
    }

}